packages/aws-cdk/lib/convert-to-cli-args.ts

```diff
@@ -0,0 +1,240 @@
+        createdBufferDays: args.createdBufferDays,
+        confirm: args.confirm,
+        bootstrapStackName: args.bootstrapStackName,
+      };
+      break;
+
+    case 'deploy':
+      commandOptions = {
+        all: args.all,
+        buildExclude: args.buildExclude,
+        exclusively: args.exclusively,
+        requireApproval: args.requireApproval,
+        notificationArns: args.notificationArns,
+        tags: args.tags,
+        execute: args.execute,
+        changeSetName: args.changeSetName,
+        method: args.method,
+        force: args.force,
+        parameters: args.parameters,
+        outputsFile: args.outputsFile,
+        previousParameters: args.previousParameters,
+        toolkitStackName: args.toolkitStackName,
+        progress: args.progress,
+        rollback: args.rollback,
+        hotswap: args.hotswap,
+        hotswapFallback: args.hotswapFallback,
+        watch: args.watch,
+        logs: args.logs,
+        concurrency: args.concurrency,
+        assetParallelism: args.assetParallelism,
+        assetPrebuild: args.assetPrebuild,
+        ignoreNoStacks: args.ignoreNoStacks,
+      };
+      break;
+
+    case 'rollback':
+      commandOptions = {
+        all: args.all,
+        toolkitStackName: args.toolkitStackName,
+        force: args.force,
+        validateBootstrapVersion: args.validateBootstrapVersion,
+        orphan: args.orphan,
+      };
+      break;
+
+    case 'import':
+      commandOptions = {
+        execute: args.execute,
+        changeSetName: args.changeSetName,
+        toolkitStackName: args.toolkitStackName,
+        rollback: args.rollback,
+        force: args.force,
+        recordResourceMapping: args.recordResourceMapping,
+        resourceMapping: args.resourceMapping,
+      };
+      break;
+
+    case 'watch':
+      commandOptions = {
+        buildExclude: args.buildExclude,
+        exclusively: args.exclusively,
+        changeSetName: args.changeSetName,
+        force: args.force,
+        toolkitStackName: args.toolkitStackName,
+        progress: args.progress,
+        rollback: args.rollback,
+        hotswap: args.hotswap,
+        hotswapFallback: args.hotswapFallback,
+        logs: args.logs,
+        concurrency: args.concurrency,
+      };
+      break;
+
+    case 'destroy':
+      commandOptions = {
+        all: args.all,
+        exclusively: args.exclusively,
+        force: args.force,
+      };
+      break;
+
+    case 'diff':
+      commandOptions = {
+        exclusively: args.exclusively,
+        contextLines: args.contextLines,
+        template: args.template,
+        strict: args.strict,
+        securityOnly: args.securityOnly,
+        fail: args.fail,
+        processed: args.processed,
+        quiet: args.quiet,
+        changeSet: args.changeSet,
+      };
+      break;
+
+    case 'metadata':
+      commandOptions = {};
+      break;
+
+    case 'acknowledge':
+      commandOptions = {};
```

Comment:
Looking at our current for this command I see:

https://github.com/aws/aws-cdk/blob/fcf4ecda4822cc8556409d4ff323e49cdbd985a7/packages/aws-cdk/lib/cli.ts#L433-L434

Should this PR also handle positional arguments? I would expect we have a strongly typed `args.acknowledge.id` or something.

Reply:
Good point. A big part of my refactor is to make sure we don't have to use `args`, `argv`, and `configuration.settings` everywhere and instead just have 1 object with arguments. 

 Currently those positional arguments are found in index 1 and beyond on `argv._`: `argv._: [Command, ...string]`

It will be a bit awkward to figure out the right way to deal with that, which I would like to do in a separate PR.